tests/conftest.py

```diff
@@ -1,15 +1,104 @@
+import shlex
+from functools import wraps
 from pathlib import Path
+from shutil import copy, copytree, which
+from subprocess import check_output
+from unittest.mock import patch
+
 import pytest
-from shutil import copytree
 
-samples_dir = Path(__file__).parent / 'samples'
+samples_dir = Path(__file__).parent / "samples"
+
+skip_if_no_git = pytest.mark.skipif(
+    (not which("git")),
+    reason="needs git to be installed and findable through the PATH environment variable",
+)
+
+
+def pytest_collection_modifyitems(items):
+    for item in items:
+        if "needgit" in item.nodeid or "needsgit" in item.nodeid:
+            item.add_marker(skip_if_no_git)
+            item.add_marker(pytest.mark.needgit)
+            item.add_marker(pytest.mark.needsgit)
+
 
 @pytest.fixture
 def copy_sample(tmp_path):
     """Copy a subdirectory from the samples dir to a temp dir"""
+
     def copy(dirname):
         dst = tmp_path / dirname
         copytree(str(samples_dir / dirname), str(dst))
         return dst
 
     return copy
+
+
+def git_cmd(repository_path, command):
+    if isinstance(command, str):
+        args = shlex.split(command)
+    else:
+        args = command
+
+    return check_output(
+        ["git", "-C", str(repository_path), *args],
+    )
+
+
+@pytest.fixture
+def tmp_git_repo(tmp_path):
+    """
+    Make a git repository in a temporary folder
+
+    The path returned is what should be passed to git's -C command, or what cwd
+    should be set to in subprocess calls
+    """
+    git_global_config = tmp_path / "git_global_config"
+    git_global_config.touch(exist_ok=False)
+    repository = tmp_path / "repository"
+    repository.mkdir(exist_ok=False)
+    with patch.dict(
+        "os.environ",
+        {
+            # https://git-scm.com/docs/git#Documentation/git.txt-codeGITCONFIGGLOBALcode
+            "GIT_CONFIG_GLOBAL": str(git_global_config),
+            # https://git-scm.com/docs/git#Documentation/git.txt-codeGITCONFIGNOSYSTEMcode
+            "GIT_CONFIG_NOSYSTEM": "true",
+            "HOME": str(tmp_path),
+            # tox by default only passes the PATH environment variable, so
+            # XDG_CONFIG_HOME is already unset
```

Comment:
If it's easy to do something appropriate with it here, we may as well not make assumptions about how tox is set up. I often run `pytest` directly for testing locally (though I don't have XDG_CONFIG_HOME set).

The comment may be more characters than overriding it. :stuck_out_tongue: 

There's also the [monkeypatch fixture](https://docs.pytest.org/en/latest/how-to/monkeypatch.html) which can be used to temporarily set & remove environment variables. I have no particular preference if you prefer to use `patch.dict`, though.

Reply:
It failed when I tried, because tox had already removed it, and I feel silly that I forgot python has try / catch 🤦

Thanks for letting me know about the monkeypatch fixture.

I personally have a preference for using features implemented by the framework, so I implemented both changes in https://github.com/takluyver/flit/pull/445/commits/a47ae163096a63cf003372eedfbd9d2c613dafc4.